### ISSUE TITLE
Intent to Implement amp-ad type: Adventive

### ISSUE THREAD

**Issue by Adventive** [ampproject/amphtml#12481]
Hello!

[Adventive](http://www.adventive.com/html5-ad-gallery/) would like to integrate our network into `amp-ad` types. PR soon :)

Also, if there are clarifications questions on implementing this, is it acceptable to ask them here, or shall I create a separate ticket for it?

Thanks!

**Comment by jridgewell** [comment]
Feel free to ask them here, or in the PR.

**Comment by lannka** [comment]
Yep. This is the place :-)

**Comment by Adventive** [comment]
Question: How `ad/ads.extern.js` come into play?

I see the majority is data attributes; from what I can gather, those fit in with the comment listed at the top of the file:
```
// HACK. Define application types used in default AMP externs
// that are not in the 3p code.
...
// Long list of, uhm, stuff the ads code needs to compile.
// All unquoted external properties need to be added here.
```
but there are also variable declarations and placeholders there. 

Any clarification on this would be helpful.

**Comment by Adventive** [comment]
@lannka, @jridgewell 
any feedback on my last question?

**Comment by Adventive** [comment]
I've updated our fork to the latest and I'm currently trying to load the locally served files on my server. I am able to load all files except I receive a `404` for `dist/ww.js` from `xhr-impls.js:163` and a related `Uncaught (in promise) Error: HTTP 404` from `log.js:489`. 

I don't see `www.js` within `dist/` so that makes sense, but the ad loads just fine via `localhost`.

Before updating to the latest `AMP_CONFIG` did not have `local_dev` set to true and was loading from the cdn but it seems #12054 fixed this?

**Comment by rsimha-amp** [comment]
@Adventive Yes, you should now see `localDev` set to `true` by default after `gulp`, `gulp watch`, `gulp build`, or `gulp dist --fortesting`.

**Comment by Adventive** [comment]
Great, 

From what I can tell, there is url replacement based on environment. (urls in `ads.amp.html` point to the cdn but in localhost are loaded locally). I also see that occurs from when loading on a non-localhost, the cdn url is replaced with my servers url.

If I point directly to localhost, it loads correctly but attempts to load the non-existent `dist/ww.js` from my server. Also none-existent on localhost. Would this be related, @rsimha-amp ?

**Comment by rsimha-amp** [comment]
I'm not familiar with the code that determines how `dist/ww.js` is loaded. @lannka?

**Comment by Adventive** [comment]
From what I've gather:
`ww.js` is a web worker file. Probably used for multiple extensions (in my case `amp-bind`)
It is dealt with in `app.js` from `904-927` (set) and `1016-1022` (get)

**Comment by rsimha-amp** [comment]
In response to the question about `ww.js`, I think the answer lies in https://github.com/ampproject/amphtml/issues/12747#issuecomment-358753920.
